### ISSUE TITLE
Evaluation code uses argmax instead of generate in prompt tuning tutorial - Prompt-based methods

### ISSUE THREAD

**Issue by maerory** [huggingface/peft#2988]
## Description
The prompt tuning tutorial's evaluation code uses `torch.argmax(outputs.logits, -1)` 
which produces gibberish outputs instead of proper text generation.

## Current Code (Incorrect)
```python
eval_preds.extend(
    tokenizer.batch_decode(
        torch.argmax(outputs.logits, -1).detach().cpu().numpy(),
        skip_special_tokens=True
    )
)
```

## Issue
`argmax` selects the most likely token at each position independently, 
without autoregressive generation. This produces incoherent output like:
```
"candedededededeausausausausausauscompl..."
```

## Proposed Fix
Use `model.generate()` for proper autoregressive text generation:
```python
generated_ids = model.generate(
    input_ids=batch["input_ids"],
    max_new_tokens=10,
    pad_token_id=tokenizer.pad_token_id
)
eval_preds.extend(
    tokenizer.batch_decode(generated_ids, skip_special_tokens=True)
)
```

## Environment
- Tutorial link: https://huggingface.co/docs/peft/task_guides/prompt_based_methods?configurations=p-tuning

## Impact
This bug affects anyone following the tutorial for classification tasks using prompt tuning with causal language models.

**Comment by maerory** [comment]
Let me know if this approach was the desired intention.

**Comment by BenjaminBossan** [comment]
Thanks for pointing this out. Indeed, the evaluation does not perform proper auto-regressive generation. I suspect that the code was written this way for speed, since the logits were already computed, but as is the `eval_preds` don't make much sense. AFAICT, they're also not used anywhere, so I think we might as well remove them completely instead of adding the code you proposed, which would slow down training. If you're interested, feel free to create a PR.

**Comment by maerory** [comment]
Sure I can do that. @BenjaminBossan
